File: ave_config.cc
/*
 * ave_config.cc
 * Copyright (C) 2026 youfa.song <vsyfar@gmail.com>
 *
 * Distributed under terms of the GPLv2 license.
 */

#include "base/ave_config.h"

namespace ave {
namespace base {

AveConfig& AveConfig::GetInstance() {
  static AveConfig config;
  return config;
}

void AveConfig::SetCodecDebugEnabled(bool enabled) {
  codec_debug_enabled_.store(enabled, std::memory_order_relaxed);
}

bool AveConfig::codec_debug_enabled() const {
  return codec_debug_enabled_.load(std::memory_order_relaxed);
}

}  // namespace base
}  // namespace ave

File: android/java/io/github/yoofa/AveConfig.java
/*
 * Copyright (C) 2026 youfa.song <vsyfar@gmail.com>
 *
 * Distributed under terms of the GPLv2 license.
 */

package io.github.yoofa;

import org.jni_zero.JniType;
import org.jni_zero.NativeMethods;

/** Process-wide AVE configuration. Set options before creating players or codecs. */
public final class AveConfig {
    private AveConfig() {}

    /** Enables codec-specific diagnostics, including the Android video input dump. */
    public static void setCodecDebugEnabled(boolean enabled) {
        AveConfigJni.get().setCodecDebugEnabled(enabled);
    }

    /** Enables verbose per-frame diagnostics in all demuxer implementations. */
    public static void setDemuxerLogsEnabled(boolean enabled) {
        AveConfigJni.get().setDemuxerLogsEnabled(enabled);
    }

    @NativeMethods
    interface Natives {
        void setCodecDebugEnabled(@JniType("bool") boolean enabled);

        void setDemuxerLogsEnabled(@JniType("bool") boolean enabled);
    }
}
